### PR TITLE
Fix default option for commands #76

### PR DESCRIPTION
**Pull request by vi4m**

### CODE REVIEW

File: Sources/Command/Run/CommandContext.swift [hunk 1]
```diff
@@ -71,6 +71,10 @@ public struct CommandContext {
 
         for opt in runnable.options {
             parsedOptions[opt.name] = try input.parse(option: opt)
+            
+            if case let CommandOptionType.value(defaultValue) = opt.type, defaultValue != nil  {
+                parsedOptions[opt.name] = defaultValue
```
Comment (calebkleveter) [inline Sources/Command/Run/CommandContext.swift]: Will this overwrite the parsed value with the default value if a default value is found?

Comment (vi4m) [reply Sources/Command/Run/CommandContext.swift]: Thanks for catching this. I provided test for this, since there was none.

**Comment by vi4m** [comment]
Done. Anything else?

**Comment by ldstreet** [comment]
If this is working as expected, maybe clearer documentation is in order. I was confused as well and expected the default value to come through when the argument was provided. I guess it depends on the most common use case. For instance, in my situation I have a `--destination` option which needs to have a default of "." and doesn't really make sense for it to be nil. I'm struggling at the moment to think of an option that could be nil, but I am sure there are some. I do think that it would be nice to work with the option with it being optional. Perhaps there is a use case for `CommandOptionType ` having the cases `.defaultedValue(default: String)`, `.value`, and `.flag`? That way when access `CommandContext` an option with type `.defaultedValue` can return a non optional value which defaults to `default`, .value can return an optional value or non optional with `requireOption`, and `.flag` can return a boolean.

**Comment by vi4m** [comment]
Not sure about the reasonable default here, so I've searched for other "prior art" at my second programming language, being Python. 

I've found  Python's implementation of "OptionParser" defaults to `not nil` when default is defined by the programmer.

Example:

zrzut ekranu 2019-01-14 o 20 28 52

zrzut ekranu 2019-01-14 o 20 27 21

**Comment by calebkleveter** [comment]
I think adding a flat to the `CommanOption.value` case so this can be customized might be better. So if you have this:

```swift
CommandOption.value(name: "number", short: "n", default: "0")
```

Not passing in the `-n` flag gives `nil`, but if you have this:

```swift
CommandOption.value(name: "number", short: "n", default: "0", defaultInAbsence: true)
```

You get `"0"`

**Comment by tanner0101** [comment]
Maybe we can rename `default` to something else? Maybe `flagValue:`?

**Pull request closed without merging**